### ISSUE TITLE
Calling functions that use a ParamSpec as arguments multiple times in another function call gives incorrect parameter errors

### ISSUE THREAD

**Issue by qiemem** [microsoft/pyright#5454]
**Describe the bug**

If you have a higher order function that uses `ParamSpec` accept the same arguments as its function, and then call that function multiple times as part of another function call, you can get incorrect parameter errors.

There are two possibly related issues here. One involves simply calling a higher order function twice. The other involves chainable higher order functions. See code examples  for both.

**To Reproduce**
1. Create a `ParamSpec` type variable
2. Create a higher order function that uses the `ParamSpec` variable for the input function and `*args`.
3. Call that function twice as arguments to another function, passing in a function that accepts at least one argument.

See code example for details.

**Expected behavior**
No parameter error

**Code or Screenshots**

***Case 1***

```python
from typing import Callable, ParamSpec

P = ParamSpec("P")


def call(fn: Callable[P, int], *args: P.args, **kwargs: P.kwargs) -> int:
    return fn(*args, **kwargs)


def plus(x: int, y: int) -> int:
    return x + y


plus(call(lambda x: x, 1), call(lambda x: x, 2))
```

Running pyright on this file produces the following error (where line 14 is the last line, and character 46 is the `2`):
```
14:46 - error: Expected 1 positional argument (reportGeneralTypeIssues)
```

***Case 2: Chaining methods***

```python
from typing import Callable, ParamSpec, Concatenate

P = ParamSpec("P")


class Pipeable:
    def __init__(self, value: int):
        self.value = value

    def pipe(
        self, fn: Callable[Concatenate[int, P], int], *args: P.args, **kwargs: P.kwargs
    ):
        return Pipeable(fn(self.value, *args, **kwargs))


dict(
    foo=Pipeable(1).pipe(lambda x, y: x + y, 2).pipe(lambda x: x+1),
    bar=Pipeable(1).pipe(lambda x, y: x + y, 2).pipe(lambda x: x+2),
)
```

This gives the following error (where line 18 is the `bar=` line, and character 9 is just after the `=`):

```
18:9 - error: Arguments for ParamSpec "P(1)@pipe" are missing
```

**VS Code extension or command-line**
Reproduced with both PyLance 2023.7.10 in VS Code and direct invocation of pyright 1.1.316 on the command line.

**Additional context**
I initially ran into this with the [Polars](https://github.com/pola-rs/polars) dataframe library, using [`Expr.pipe`](https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.pipe.html) for multiple arguments in an [`GroupBy.agg`](https://pola-rs.github.io/polars/py-polars/html/reference/dataframe/api/polars.dataframe.groupby.GroupBy.agg.html#polars.dataframe.groupby.GroupBy.agg).

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.317, which I just published. It will also be included in future releases of pylance.
